src/object/interrupt.c

```diff
@@ -145,12 +145,20 @@ void invokeIRQHandler_AckIRQ(irq_t irq)
 
 #if defined ENABLE_SMP_SUPPORT && defined CONFIG_ARCH_ARM
     if (IRQ_IS_PPI(irq) && IRQT_TO_CORE(irq) != getCurrentCPUIndex()) {
+#ifdef CONFIG_ARM_GIC_V3_SUPPORT
+        doRemoteDeactivatePrivateInterrupt(IRQT_TO_CORE(irq), IRQT_TO_IDX(irq));
+#else /* CONFIG_ARM_GIC_V3_SUPPORT */
         doRemoteMaskPrivateInterrupt(IRQT_TO_CORE(irq), false, IRQT_TO_IDX(irq));
+#endif /* CONFIG_ARM_GIC_V3_SUPPORT */
         return;
     }
 #endif
+#ifdef CONFIG_ARM_GIC_V3_SUPPORT
+    deactivateInterrupt(irq);
+#else /* CONFIG_ARM_GIC_V3_SUPPORT */
     maskInterrupt(false, irq);
-#endif
+#endif /* CONFIG_ARM_GIC_V3_SUPPORT */
```

Comment:
If the `endif` is this close to the `if` I would drop the comment as it only makes things messy and it's clear enough without it.